src/test/ui/lifetimes/issue-105227.rs

```diff
@@ -0,0 +1,18 @@
+fn chars0(v :(& str, &str)) -> impl Iterator<Item = char> {
+//~^ HELP
+//~| HELP
```

Comment:
Thanks for reviewing the code. I've updated the test code with gist's. 
But with run-rustfix enabled, the test fails because rustfix provides the following fix which is incorrect:
```
fn chars0<'a><'a>(v :(&'a 'a  str, &'a 'a str)) -> impl Iterator<Item = char> + 'a  + 'a  {
```
Here is the outputs of the test which `--bless` with `// run-restfix` enabled:
```
Actual fixed saved to /home/ychen/rust-new/rust/build/x86_64-unknown-linux-gnu/test/ui/lifetimes/issue-105227/issue-105227.fixed
Actual fixed saved to /home/ychen/rust-new/rust/src/test/ui/lifetimes/issue-105227.fixed

error: failed to compile fixed code
status: exit status: 1
command: "/home/ychen/rust-new/rust/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/home/ychen/rust-new/rust/src/test/ui/lifetimes/issue-105227.fixed" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "-o" "/home/ychen/rust-new/rust/build/x86_64-unknown-linux-gnu/test/ui/lifetimes/issue-105227/a" "-Crpath" "-Cdebuginfo=0" "-Lnative=/home/ychen/rust-new/rust/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/home/ychen/rust-new/rust/build/x86_64-unknown-linux-gnu/test/ui/lifetimes/issue-105227/auxiliary"
stdout: none
--- stderr -------------------------------
error: expected `(`, found `<`
  --> /home/ychen/rust-new/rust/src/test/ui/lifetimes/issue-105227.fixed:5:14
   |
LL | fn chars0<'a><'a>(v :(&'a 'a  str, &'a 'a str)) -> impl Iterator<Item = char> + 'a  + 'a  {
   |              ^ expected `(`

error: aborting due to previous error
------------------------------------------



failures:
    [ui] src/test/ui/lifetimes/issue-105227.rs
```

How do we fix the rustfix problem?

Reply:
It seems to apply the suggestion twice, because the error is giving the help twice, once for each parameter. One could only emit the suggestion once per occurence, but not sure if that's an advantage. People might only read one of the errors, and not see the help. I guess it's okay to not add rustfix to the tests.